api/uaa/auth.go

```diff
@@ -18,17 +19,23 @@ type AuthResponse struct {
 
 // Authenticate sends a username and password to UAA then returns an access
 // token and a refresh token.
-func (client Client) Authenticate(ID string, secret string, grantType constant.GrantType) (string, string, error) {
+func (client Client) Authenticate(ID string, secret string, origin string, grantType constant.GrantType) (string, string, error) {
 	requestBody := url.Values{
 		"grant_type": {string(grantType)},
 	}
 	switch grantType {
 	case constant.GrantTypeClientCredentials:
 		requestBody.Set("client_id", ID)
 		requestBody.Set("client_secret", secret)
+		if origin != "" {
+			return "", "", errors.New("Incorrect Usage: The following arguments cannot be used together: --client-credentials, --origin")
+		}
 	default:
 		requestBody.Set("username", ID)
 		requestBody.Set("password", secret)
+		if origin != "" {
+			requestBody.Set("login_hint", "{\"origin\":\""+origin+"\"}")
```

Comment:
Per the [UAA docs](http://docs.cloudfoundry.org/api/uaa/version/4.19.0/index.html#authorization), this needs to be a query parameter with a value of URL encoded JSON. Thus, when creating the request (which is below) with `newRequest`, include the `Query` struct field in the `requestOptions`.

Additionally there should be a corresponding test validation for this change (not just passing the value into the function header).